### PR TITLE
Add consistent API query parameter validation

### LINKED ISSUE #9193: [api] Perform robust and consistent query param validation
Our API is large and accepts many query parameters. Occasionally we have an issue with the UI sending an unsupported param, or a param with an invalid value. Handling on the server side is a bit haphazard, sometimes silently ignoring, sometimes returning status errors, etc.

This has not been a huge issue, but with the API possibly not being accessed by more clients (AI, etc), it may be time to improve in this area.

All API calls should validate that the supplied query params are supported, and if supported, are set with a valid value. Otherwise we should likely return a 409. I think in the cases that we do something similar today, we are passing back 400, which isn't bad, but 409 is probably a bit better.

### PR DESCRIPTION
**Pull request by pbajjuri20**
Introduce handlers/queryparams shared helpers to reject unsupported query parameters and validate supported values across REST handlers. Return HTTP 400 for query param errors via RespondWithQueryParamError.

Fixes kiali/kiali#9193

@pbajjuri20 Please remember to remove the PR boilerplate after providing sufficient detail. I removed it.

### CODE REVIEW

**Comment by pbajjuri20** [comment]
@coderabbitai review

**Comment by pbajjuri20** [comment]
@jshaughn: At last, CI passed successfully.
Please review this MR when you have some time.

File: handlers/queryparams/queryparams.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	"net/url"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/kiali/kiali/config"
+)
+
+// ErrorStatusCode is the HTTP status returned for invalid or unsupported query parameters.
+const ErrorStatusCode = http.StatusConflict
```
Comment (jshaughn) [inline handlers/queryparams/queryparams.go]: See my comment in the issue, but I think the suggestion to use this status code was a bad idea. We should just use 400 (bad request). Or maybe a combination of 400 and 422, up to you.

File: handlers/apps.go [hunk 2]
```diff
@@ -31,22 +31,25 @@ type appParams struct {
...
-func (p *appParams) extract(r *http.Request, conf *config.Config) {
+func (p *appParams) extract(r *http.Request, conf *config.Config) error {
 	vars := mux.Vars(r)
 	query := r.URL.Query()
-	p.baseExtract(conf, r)
+	if err := p.baseExtract(conf, r, "clusterName", "health", "istioResources", "namespaces", "queryTime", "rateInterval"); err != nil {
+		return err
+	}
 	p.Namespace = vars["namespace"]
-	p.ClusterName = clusterNameFromQuery(conf, query)
```
Comment (jshaughn) [inline handlers/apps.go:39]: Removal of `handlers/utils.go#clusterNameFromQuery` is good here, but I see that the function still exists and is still used in various places.  I think it should probably be removed completely in favor of the new function in queryparams.

File: handlers/queryparams/queryparams.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+// validPromDurationRe matches a valid Prometheus duration string (e.g. "5m", "1h30m").
+var validPromDurationRe = regexp.MustCompile(`^([0-9]+(ms|s|m|h|d|w|y))+$`)
```
Comment (jshaughn) [inline handlers/queryparams/queryparams.go:18]: good change of the regex, more robust.

File: handlers/queryparams/queryparams.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+// ClusterName extracts the cluster name from query parameters, defaulting to the configured cluster.
+func ClusterName(conf *config.Config, query url.Values) string {
+	cluster := query.Get("clusterName")
+	if cluster == "" {
+		cluster = conf.KubernetesConfig.ClusterName
+	}
+	return cluster
+}
```
Comment (jshaughn) [inline handlers/queryparams/queryparams.go:44]: As mentioned elsewhere, this should be used everywhere, the old utility should be removed.

File: handlers/queryparams/tracing_test.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kiali/kiali/config"
+)
+
+func TestParseTracingQueryUnknownParam(t *testing.T) {
```
Comment (jshaughn) [inline handlers/queryparams/tracing_test.go:15]: Could add more unit testing here: No tests for:

Valid parsing (timestamps, durations, tags, limits)
Boundary values (negative numbers, overflows)
tags=null handling
QueryScope injection from config

**Review by jshaughn — COMMENTED** [review]
@pbajjuri20 Good work so far, I like the new queryparams package with the consolidated parameter functions for consistent handling. I've made a few comments but my main comment is that I think this can go farther.

I  don't really love how the query param "allowlists" are built into the arg list for RejectUnknown. It sort of has two issues:
- Duplication: each param name appears in both the RejectUnknown allowlist AND in the manual parsing code that follows. Add a param to one but not the other and you can get a bug.
- Scattered validation — the type/default info is spread across the handler body rather than declared up front.

I'd suggest having a convention where each Handler explicitly defined its params and possibly their types and default values, and then just pass it all into the query param parser/validator. Special case handling could still be done after that, if needed.

Here is an example of what I'm talking about (I asked AI to generate a sample based on my comment...), Doesn't have to be exactly like this, but the idea is to be more declarative about each param, so it's easy to see what is accepted up front, and to handle most validation in one shot. so something like:

```
// Package-level schema declaration — serves as documentation of the endpoint contract.
var appDetailsParams = []queryparams.Param{
    queryparams.BoolParam("health", true),
    queryparams.BoolParam("istioResources", true),
    queryparams.ClusterParam(),
    queryparams.StringParam("namespaces", ""),
    queryparams.TimestampParam("queryTime"),
    queryparams.PromDurationParam("rateInterval", config.DefaultHealthRateInterval),
}

// In handler:
func AppDetails(...) http.HandlerFunc {
    return func(w http.ResponseWriter, r *http.Request) {
        result, err := queryparams.ParseWithConfig(r.URL.Query(), conf, appDetailsParams)
        if err != nil {
            RespondWithQueryParamError(w, err.Error())
            return
        }

        criteria := business.AppCriteria{
            IncludeHealth:         result.Bool("health"),
            IncludeIstioResources: result.Bool("istioResources"),
            Cluster:              result.String("clusterName"),
            RateInterval:         result.Duration("rateInterval"),
            QueryTime:            result.Time("queryTime"),
        }
        // ...
    }
}
```
We can talk about it more if you like.

one small nit: please don't leave boilerplate in the PR description.

File: handlers/services.go [hunk 6]
```diff
@@ -32,24 +32,42 @@ type serviceListParams struct {
 	IncludeOnlyDefinitions bool `json:"onlyDefinitions"`
 }
 
-func (p *serviceListParams) extract(conf *config.Config, r *http.Request) {
+// Package-level schema — documents the service list query contract.
```
Comment (jshaughn) [inline handlers/services.go:35]: AI sometimes likes to add these sort of obvious comments. Try to avoid comments that don't really add anything. Not a blocker, just mentioning it as a cleanup for the future.

**Review by jshaughn — APPROVED** [review]
Very good changes, @pbajjuri20 . I added the `ci-full` label, something like this should pass the full set of test jobs to try and ensure the API changes don't break anything. If CI all passes this should be good.

When merging, just remember to sqash-merge **AND** make sure you update the commit comment to something sane. You can ask AI to generate a good final commit comment for the PR and just paste that in.

File: handlers/services.go (hunk 6 above)
Comment (pbajjuri20) [reply handlers/services.go:35]: Thanks @jshaughn, for your review. 

Next time, I will verify completely.
